### PR TITLE
feat(sdk): multipart content

### PR DESCRIPTION
**Pull request by ShookLyngs**
## Changes
- Add multipart content relevant utilities/tests
  - Utilities: `encodeMultipartContent`, `decodeMultipartContent`  
  - Tests: `MultipartContent.test.ts`

### Multipart content

This is a MIME implementation that enables users to put multiple formats of data into one `SporeData.content`.

A simple example of multipart content in spore:

SporeData.contentType

```
multipart/mixed;boundary="simple boundary"
```

SporeData.content

```
This is the preamble.  It is to be ignored, though it is a handy place for mail composers to include an explanatory note to non-MIME compliant readers. 
--simple boundary 

This is implicitly typed plain ASCII text. It does NOT end with a linebreak. 
--simple boundary 
Content-type: text/plain; charset=us-ascii 

This is explicitly typed plain ASCII text. It DOES end with a linebreak. 

--simple boundary-- 
This is the epilogue.  It is also to be ignored.
```

## Reference
- https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html
- https://www.ietf.org/rfc/rfc2046.html#section-5.1

### CODE REVIEW

File: packages/core/src/helpers/multipartContent.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+export type AsyncableIterable<T> = AsyncIterable<T> | Iterable<T>;
+
+export type ResolvedMultipartContent = {
+  headers: Headers;
+  body?: Uint8Array | null;
+  parts?: ResolvedMultipartContent[] | null;
+};
+
+export async function decodeMultipartContent(message: string, boundary: string): Promise<ResolvedMultipartContent[]> {
```
Comment (code-monad) [inline packages/core/src/helpers/multipartContent.ts:16]: If there's no boundary text included, what will happen?

Comment (ShookLyngs) [reply packages/core/src/helpers/multipartContent.ts:16]: > If there's no boundary text included, what will happen?

Do you mean like this? The below example will just throws an "invalid message" error:

```ts
decodeMultipartContent('text without boundary', 'boundary')
```

Comment (code-monad) [reply packages/core/src/helpers/multipartContent.ts:16]: Shall there be a detector func to check whether this is a valid multipart?

Comment (ShookLyngs) [reply packages/core/src/helpers/multipartContent.ts:16]: > Shall there be a detector func to check whether this is a valid multipart content data?

Added simple APIs for testing decodability: `isMultipartContentValid`, `isMultipartContentAsBytesValid`

**Review by Dawn-githup — APPROVED** [review]

**Review by code-monad — APPROVED** [review]

**Pull request merged**